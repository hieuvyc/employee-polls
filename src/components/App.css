.App {
  text-align: center;
}

.login-container {
  text-align: center;
  max-width: 400px;
  margin: 0 auto;
  padding: 50px 20px;
}

.avatar {
  border-radius: 50%;
}

.form-group {
  margin-bottom: 15px;
}

input {
  width: 100%;
  padding: 10px;
  margin-bottom: 10px;
  border: 1px solid #ccc;
  border-radius: 4px;
}

.btn {
  background-color: #4CAF50;
  color: white;
  padding: 10px 20px;
  border: none;
  cursor: pointer;
  width: 100%;
  border-radius: 4px;
}

.btn:disabled {
  background-color: #ccc;
  cursor: not-allowed;
}

