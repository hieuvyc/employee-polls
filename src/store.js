import {configureStore} from '@reduxjs/toolkit';
import authedUser from "./reducers/authedUser";
import users from "./reducers/users";
import questions from "./reducers/questions";
import logger from "./middlewave/logger";

export const store = configureStore({
    reducer: {
        authedUser,
        users,
        questions,
    },
    middleware: (getDefaultMiddleware) =>
        getDefaultMiddleware().concat(logger),
});
