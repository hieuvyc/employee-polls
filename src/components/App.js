import React, { useEffect } from 'react';
import { connect } from 'react-redux';
import { Routes, Route, Navigate } from 'react-router-dom';
import { handleInitialData } from '../actions/shared';
import LoginPage from './LoginPage';
import Dashboard from './Dashboard';
import PollPage from './PollPage';
import PollCreationPage from './PollCreationPage';
import LeaderboardPage from './LeaderboardPage';
import NavBar from './NavBar';
import '../css/App.css';
import NotFound from "./NotFound";

const App = (props) => {
  const { authedUser, loading, dispatch } = props;

  useEffect(() => {
    dispatch(handleInitialData());
  }, [dispatch]);

  return (
      <div className="App">
        {!authedUser ? (
            <Routes>
              <Route path="*" element={<LoginPage />} />
            </Routes>
        ) : (
            <>
              <NavBar />
              {loading ? null : (
                  <Routes>
                    <Route path="/" element={<Dashboard />} />
                    <Route path="/questions/:question_id" element={<PollPage />} />
                    <Route path="/add" element={<PollCreationPage />} />
                    <Route path="/leaderboard" element={<LeaderboardPage />} />
                    <Route path="*" element={<Navigate to="/" />} />
                    <Route path="/not-found" exact element={<NotFound/>}/>
                  </Routes>
              )}
            </>
        )}
      </div>
  );
};

const mapStateToProps = ({ authedUser }) => ({
  authedUser,
  loading: authedUser === null,
});

export default connect(mapStateToProps)(App);
