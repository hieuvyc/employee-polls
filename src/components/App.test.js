import React from 'react';
import {render} from '@testing-library/react';
import App from './App';
import {Provider} from "react-redux";
import {store} from "../store";
import {BrowserRouter} from "react-router-dom";

describe("App", () => {
  it("should render the App component and match the snapshot", () => {
    const component = render(
        <Provider store={store}>
          <BrowserRouter>
            <App/>
          </BrowserRouter>
        </Provider>
    );
    expect(component).toBeDefined();
    expect(component).toMatchSnapshot();
  });
});
