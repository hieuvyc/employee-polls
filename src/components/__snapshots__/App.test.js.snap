// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render the App component and match the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <div
          class="login-container"
        >
          <h1>
            Employee Polls
          </h1>
          <img
            alt="Employee Avatars"
            class="avatar"
            src="https://www.polly.ai/hubfs/Blog%20Images/Illustrations%20(white,%20svg)/Employee%20Feedback%202.svg"
            style="width: 400px; height: 400px;"
          />
          <h3>
            Log In
          </h3>
          <form>
            <div
              class="form-group"
            >
              <input
                placeholder="User"
                required=""
                type="text"
                value="sarahedo"
              />
            </div>
            <div
              class="form-group"
            >
              <input
                placeholder="Password"
                required=""
                type="password"
                value="password123"
              />
            </div>
            <button
              class="btn"
              type="submit"
            >
              Submit
            </button>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <div
        class="login-container"
      >
        <h1>
          Employee Polls
        </h1>
        <img
          alt="Employee Avatars"
          class="avatar"
          src="https://www.polly.ai/hubfs/Blog%20Images/Illustrations%20(white,%20svg)/Employee%20Feedback%202.svg"
          style="width: 400px; height: 400px;"
        />
        <h3>
          Log In
        </h3>
        <form>
          <div
            class="form-group"
          >
            <input
              placeholder="User"
              required=""
              type="text"
              value="sarahedo"
            />
          </div>
          <div
            class="form-group"
          >
            <input
              placeholder="Password"
              required=""
              type="password"
              value="password123"
            />
          </div>
          <button
            class="btn"
            type="submit"
          >
            Submit
          </button>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`App should render the component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <div
          class="login-container"
        >
          <h1>
            Employee Polls
          </h1>
          <img
            alt="Employee Avatars"
            class="avatar"
            src="https://www.polly.ai/hubfs/Blog%20Images/Illustrations%20(white,%20svg)/Employee%20Feedback%202.svg"
            style="width: 400px; height: 400px;"
          />
          <h3>
            Log In
          </h3>
          <form>
            <div
              class="form-group"
            >
              <input
                placeholder="User"
                required=""
                type="text"
                value="sarahedo"
              />
            </div>
            <div
              class="form-group"
            >
              <input
                placeholder="Password"
                required=""
                type="password"
                value="password123"
              />
            </div>
            <button
              class="btn"
              type="submit"
            >
              Submit
            </button>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <div
        class="login-container"
      >
        <h1>
          Employee Polls
        </h1>
        <img
          alt="Employee Avatars"
          class="avatar"
          src="https://www.polly.ai/hubfs/Blog%20Images/Illustrations%20(white,%20svg)/Employee%20Feedback%202.svg"
          style="width: 400px; height: 400px;"
        />
        <h3>
          Log In
        </h3>
        <form>
          <div
            class="form-group"
          >
            <input
              placeholder="User"
              required=""
              type="text"
              value="sarahedo"
            />
          </div>
          <div
            class="form-group"
          >
            <input
              placeholder="Password"
              required=""
              type="password"
              value="password123"
            />
          </div>
          <button
            class="btn"
            type="submit"
          >
            Submit
          </button>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
